src/main/java/org/candlepin/subscriptions/retention/TallyRetentionController.java

```diff
@@ -63,8 +76,17 @@ public void cleanStaleSnapshotsForAccount(String accountNumber) {
             if (cutoffDate == null) {
                 continue;
             }
-            repository.deleteAllByAccountNumberAndGranularityAndSnapshotDateBefore(accountNumber,
+            tallySnapshotRepository.deleteAllByAccountNumberAndGranularityAndSnapshotDateBefore(accountNumber,
                 granularity, cutoffDate);
         }
     }
+
+    public void purgeOldEventRecords() {
+        Duration eventRetentionDuration = eventRecordsRetentionProperties.getEventRetentionDuration();
+
+        log.info("Purging event records older than Duration {}", eventRetentionDuration);
```

Comment:
I would suggest maybe changing this to something like:

```
OffsetDateTime cutoff = OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS).minus(eventRetentionDuration);
log.info("Purging all event records prior to {}", cutoff);
eventRecordRepository.deleteEventRecordsByTimestampBefore(cutoff);
```

since the truncation to `ChronoUnit.DAYS` means that if you tried to do the math yourself by looking at the log entry timestamp and the value of the Duration, it wouldn't add up.